cmd/erasure-object.go

```diff
@@ -1153,9 +1153,7 @@ func (er erasureObjects) DeleteObjects(ctx context.Context, bucket string, objec
 			errs[objIndex] = toObjectErr(err, bucket, objects[objIndex].ObjectName)
 		}
 
-		if errs[objIndex] == nil {
-			defer NSUpdated(bucket, objects[objIndex].ObjectName)
-		}
+		defer NSUpdated(bucket, objects[objIndex].ObjectName)
```

Comment:
why should we call NSUpdated when the object was not successfully deleted?

Reply:
Its on purpose to make sure that we run scanner at this path when delete failed - to ensure we can heal in-case @krisis